user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserOperator.java

```diff
@@ -112,12 +119,16 @@ private final String getLockName(String namespace, String name) {
      * @param handler Completion handler
      */
     protected void createOrUpdate(Reconciliation reconciliation, KafkaUser kafkaUser, Secret clientsCaCert, Secret clientsCaKey, Secret userSecret, Handler<AsyncResult<Void>> handler) {
+        Future<Void> createOrUpdateFuture = Future.future();
         String namespace = reconciliation.namespace();
         String userName = reconciliation.name();
         KafkaUserModel user;
+        KafkaUserStatus userStatus = new KafkaUserStatus();
         try {
             user = KafkaUserModel.fromCrd(certManager, passwordGenerator, kafkaUser, clientsCaCert, clientsCaKey, userSecret);
         } catch (Exception e) {
+            StatusUtils.setStatusConditionAndObservedGeneration(kafkaUser, userStatus, Future.failedFuture(e));
+            updateStatus(kafkaUser, reconciliation, userStatus);
```

Comment:
I think we should handle the result here. This way it doesn't sync properly and you might have another reconciliation started while still updating the status here. You should probably handle the handler with the original exception only when the status update completes.